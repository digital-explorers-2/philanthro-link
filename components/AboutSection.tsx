import React from 'react';
import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "@/components/ui/collapsible";

function AboutSection() {
  return (
    <div className="p-6 bg-gray-100">
      <div className="mt-16 ml-auto mr-auto max-w-[970px]">
        <p className="text-sm text-gray-500 mb-4">MODERN CROWDFUNDING PLATFORM</p>
        <h2 className="text-3xl font-bold font-dm-sans mb-10">
          <span className="text-black">Distribute aid</span>
          <span className="text-primary"> easily</span>
          <span className="text-black">,</span>
          <span className="text-primary"> quickly</span>
          <span className="text-black">, and</span>
          <br />
          <span className="text-primary"> transparently</span>
          <span className="text-black">.</span>
        </h2>

        <div className="flex items-center">
          <img src="/phone.png" alt="Phone" className="w-screen-phone mr-4" />
          <div className="ml-36">
            <p className="text-gray-600 mb-4">
              PhilanthroLink is a crowdfunding platform dedicated to connecting donors with causes they care about. It provides a seamless way for individuals and organizations to donate to verified charitable projects while fostering a sense of community and collaboration in the realm of philanthropy.
            </p>
            <a href="#" className="text-black underline">Read more</a>
            <h3 className="text-2xl font-bold mt-16 mb-8">Frequently asked questions</h3>

            {/* Collapsible 1 */}
            <div className="bg-white shadow-md rounded-lg mb-4 p-4">
              <Collapsible>
                <CollapsibleTrigger className="font-semibold text-primary cursor-pointer my-2">
                  Lorem ipsum sit dolor?
                </CollapsibleTrigger>
                <CollapsibleContent>
                  <span className="text-primary">Answer to question 1 goes here.</span>
                </CollapsibleContent>
              </Collapsible>
            </div>

            {/* Collapsible 2 */}
            <div className="bg-white shadow-md rounded-lg mb-4 p-4">
              <Collapsible>
                <CollapsibleTrigger className="font-semibold text-primary cursor-pointer my-2">
                  Lorem ipsum sit dolor?
                </CollapsibleTrigger>
                <CollapsibleContent>
                  <span className="text-primary">Answer to question 2 goes here.</span>
                </CollapsibleContent>
              </Collapsible>
            </div>

            {/* Collapsible 3 */}
            <div className="bg-white shadow-md rounded-lg mb-4 p-4">
              <Collapsible>
                <CollapsibleTrigger className="font-semibold text-primary cursor-pointer my-2">
                  Lorem ipsum sit dolor?
                </CollapsibleTrigger>
                <CollapsibleContent>
                  <span className="text-primary">Answer to question 3 goes here.</span>
                </CollapsibleContent>
              </Collapsible>
            </div>
          </div>
        </div>
        
      </div>
    </div>
  );
}

export default AboutSection;
