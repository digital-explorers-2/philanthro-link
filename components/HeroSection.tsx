import React from 'react';

export default function HeroSection() {
  return (
    <div className="relative flex flex-col items-center justify-center min-h-screen px-6 sm:px-12 md:px-24 lg:px-32 xl:px-48 py-12">
      <h1 className="text-5xl font-bold text-center font-dm-sans text-primary mb-8">
        <span>Happiness</span>
        <span className="text-white"> comes from</span>
         <br />
        <span>your action.</span>
      </h1>
      <p className="text-base font-normal text-center text-white mb-8">
        Be a part of the breakthrough and make someone's dream come true.
      </p>
      <img src="/hero-bg.jpg" alt="Hero" className="absolute inset-0 object-cover w-full h-full brightness-50 z-[-1]" />
      <div className="flex mt-4">
        <button className="w-[180px] h-[55px] px-8 bg-primary rounded-full text-white mr-4">
          Donate Now
        </button>
        <button className="w-[180px] h-[55px] px-8 bg-transparent rounded-full border hover:bg-primary border-gray-300 text-white">
          About Us
        </button>
      </div>
    </div>
  );
}
