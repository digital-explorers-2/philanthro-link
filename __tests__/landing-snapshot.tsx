import { render } from "@testing-library/react";
import AboutSection from "@/components/AboutSection";
import HeroSection from "@/components/HeroSection";

describe("Landing page snapshot unchanged", () => {
  it("renders hero section", () => {
    const { container } = render(<HeroSection />);
    expect(container).toMatchSnapshot();
  });
  it("renders about section", () => {
    const { container } = render(<AboutSection />);
    expect(container).toMatchSnapshot();
  });
});
