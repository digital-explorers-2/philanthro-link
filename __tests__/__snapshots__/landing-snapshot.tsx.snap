// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page snapshot unchanged renders about section 1`] = `
<div>
  <div
    class="p-6 bg-gray-100"
  >
    <div
      class="mt-4 mx-auto max-w-[970px]"
    >
      <p
        class="text-sm text-gray-500 mb-4 mt-10"
      >
        MODERN CROWDFUNDING PLATFORM
      </p>
      <h2
        class="text-3xl font-bold font-dm-sans mb-10"
      >
        <span
          class="text-black"
        >
          Distribute aid
        </span>
        <span
          class="text-primary"
        >
           easily, quickly,
        </span>
        <br />
        <span
          class="text-primary"
        >
           and transparently.
        </span>
      </h2>
      <div
        class="flex flex-col md:flex-row items-center"
      >
        <img
          alt="Phone"
          class="w-screen-phone mb-4 md:mr-4 md:mb-0 pr-6 hidden md:block"
          src="/phone.png"
        />
        <div
          class="md:ml-4"
        >
          <p
            class="text-gray-600 mb-4 md:text-left"
          >
            PhilanthroLink is a crowdfunding platform dedicated to connecting donors with causes they care about. It provides a seamless way for individuals and organizations to donate to verified charitable projects while fostering a sense of community and collaboration in the realm of philanthropy.
          </p>
          <h3
            class="text-2xl font-bold mt-8 mb-4 text-center md:text-left pb-6"
          >
            Frequently asked questions
          </h3>
          <div
            class="w-full"
            data-orientation="vertical"
          >
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r1:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r0:"
                  type="button"
                >
                  How can I donate through PhilanthroLink?
                  <svg
                    class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r0:"
                class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r1:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r3:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r2:"
                  type="button"
                >
                  Are the charitable projects on PhilanthroLink verified?
                  <svg
                    class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r2:"
                class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r3:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r5:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r4:"
                  type="button"
                >
                  Can I create a fundraising campaign on PhilanthroLink?
                  <svg
                    class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r4:"
                class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r5:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r7:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r6:"
                  type="button"
                >
                  How can I get in touch with PhilanthroLink?
                  <svg
                    class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r6:"
                class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r7:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r9:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r8:"
                  type="button"
                >
                  Can I volunteer with PhilanthroLink?
                  <svg
                    class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r8:"
                class="overflow-hidden text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r9:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
          </div>
        </div>
      </div>
      <p
        class="text-sm text-gray-500 mb-4 mt-16"
      >
        HUMANITARIAN MISSION
      </p>
      <h2
        class="text-3xl font-bold font-dm-sans mb-10 md:text-left"
      >
        <span
          class="text-black"
        >
          Help the affected by
        </span>
        <br />
        <span
          class="text-primary"
        >
           disaster, shortage,
        </span>
        <br />
        <span
          class="text-primary"
        >
           and emergency relief.
        </span>
      </h2>
      <div
        class="flex flex-col md:flex-row items-center"
      >
        <div
          class="md:mr-4 md:mb-0"
        >
          <p
            class="text-gray-600 mb-4 md:text-left"
          >
            <span
              class="text-primary"
            >
              22,690 
            </span>
            Donations have been verified and still active.
          </p>
          <p
            class="text-gray-600 mb-4 md:text-left"
          >
            <span
              class="text-primary"
            >
              6,450 
            </span>
            Donations have been distributed to disaster-affected areas.
          </p>
          <p
            class="text-gray-600 mb-4 md:text-left"
          >
            <span
              class="text-primary"
            >
              1.4 Billion 
            </span>
            total funds raised so far.
          </p>
          <p
            class="text-gray-600 mb-4 md:text-left"
          >
            <span
              class="text-primary"
            >
              10,517 
            </span>
            donations have been distributed to the needy.
          </p>
          <p
            class="text-gray-600 mb-4 md:text-left"
          >
            <span
              class="text-primary"
            >
              5,058 
            </span>
            donations were distributed to social foundations and orphanages.
          </p>
          <p
            class="text-gray-600 mb-4 md:text-left"
          >
            <span
              class="text-primary"
            >
              4,803 
            </span>
            donations have been distributed to people in emergency situations.
          </p>
        </div>
        <img
          alt="About Circle"
          class="w-[314px] h-[331px] mt-10 md:mt-0 md:ml-auto"
          src="/about-circle.png"
        />
      </div>
      <div
        class="hidden md:block shadow-md rounded-2xl mb-4 mt-20"
      >
        <img
          alt="Contact Banner"
          class="w-full h-[220px]"
          src="/contact-banner.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Landing page snapshot unchanged renders hero section 1`] = `
<div>
  <div
    class="relative flex flex-col items-center justify-center min-h-screen px-6 sm:px-12 md:px-24 lg:px-32 xl:px-48 py-8"
  >
    <h1
      class="text-4xl md:text-5xl font-bold text-center font-dm-sans text-primary mb-4 md:mb-8"
    >
      <span>
        Happiness
      </span>
      <span
        class="text-white"
      >
         comes from
      </span>
      <br />
      <span>
        your action.
      </span>
    </h1>
    <p
      class="text-base md:text-lg font-normal text-center text-white mb-4 md:mb-8"
    >
      Be a part of the breakthrough and make someone's dream come true.
    </p>
    <img
      alt="Hero"
      class="absolute inset-0 object-cover w-full h-full brightness-[0.3] z-[-1]"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=%2Fhero-bg.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fhero-bg.jpg&w=640&q=75 640w, /_next/image?url=%2Fhero-bg.jpg&w=750&q=75 750w, /_next/image?url=%2Fhero-bg.jpg&w=828&q=75 828w, /_next/image?url=%2Fhero-bg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fhero-bg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fhero-bg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fhero-bg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fhero-bg.jpg&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <div
      class="flex flex-row mt-4 gap-4"
    >
      <a
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 rounded-full px-10"
        href="#donations"
      >
        Donate now
      </a>
      <a
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input hover:bg-accent h-10 py-2 rounded-full px-10 bg-transparent text-white hover:text-black"
        href="/about"
      >
        About us
      </a>
    </div>
  </div>
</div>
`;
